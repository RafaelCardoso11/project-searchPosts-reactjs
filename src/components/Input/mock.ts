import { InputType } from "./types";
const fn = jest.fn();

export const inputPropsMock:InputType  = {
 placeHolder: 'place holder',
 type: 'text',
 value: 'value',
 event: fn
}
