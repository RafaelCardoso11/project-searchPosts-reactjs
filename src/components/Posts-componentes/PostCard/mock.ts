import { Post } from '../types';

export const postCardPropsMock: Post = {
  id: 1,
  title: 'title 1',
  body: 'body 1',
  cover: 'img/img.png',
  alt: 'title 1',
};
