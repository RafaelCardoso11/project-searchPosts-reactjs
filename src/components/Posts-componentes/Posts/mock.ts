import { Post } from "../types";

export const postsPropsMock: Post[] = [
  {
    id: 1,
    title: "title 1",
    body: "body 1",
    cover: "img/img1.png",
    alt: "title 1",
  },
  {
    id: 2,
    title: "title 2",
    body: "body 2",
    cover: "img/img2.png",
    alt: "title 2",
  },
  {
    id: 3,
    title: "title 3",
    body: "body 3",
    cover: "img/img3.png",
    alt: "title 3",
  },
];