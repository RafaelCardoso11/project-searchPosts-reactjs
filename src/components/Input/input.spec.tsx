import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Input } from ".";
import { inputPropsMock } from "./mock";

const props = inputPropsMock;
describe("<Input />", () => {
  it("should have a value of searchValue", () => {
    render(<Input {...props} />);
    const input: any = screen.getByPlaceholderText(props.placeHolder);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe(props.value);
  });
  it("should call handleChange function on each pressed", () => {
    render(<Input {...props} />);

    const input: any = screen.getByPlaceholderText(props.placeHolder);
    const value = props.value;

    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(props.event).toHaveBeenCalledTimes(value.length);
    expect(input).toHaveAttribute("type", props.type);
  });
  it("should match snapshot", () => {
    const { container } = render(<Input {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
