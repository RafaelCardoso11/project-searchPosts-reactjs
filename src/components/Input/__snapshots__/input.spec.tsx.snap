// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<input
  class="input"
  placeholder="place holder"
  type="text"
  value="value"
/>
`;
